jest.setup.js

```diff
@@ -1,7 +1,24 @@
 // Mock Web Media APIs
 
-global.MediaStream = jest.fn();
-global.MediaStream.prototype = {
-  getTracks: jest.fn(() => []),
-  removeTrack: jest.fn(),
-};
+global.MediaStream = jest.fn((tracksOrStream = []) => {
+  let tracks;
+
+  if (tracksOrStream instanceof global.MediaStream) {
+    tracks = [...tracksOrStream.getTracks()];
+  } else {
+    tracks = tracksOrStream;
+  }
+
+  for (const track of tracks) {
+    if (!('enabled' in track)) {
+      track.enabled = true;
+    }
+  }
+
+  return {
+    getTracks: jest.fn(() => tracks),
+    getAudioTracks: jest.fn(() => tracks.filter((track) => track.kind === 'audio')),
+    getVideoTracks: jest.fn(() => tracks.filter((track) => track.kind === 'video')),
+    removeTrack: jest.fn(),
```

Comment:
you can make this work by using an in-place filtering to exclude the received track from the tracks array https://stackoverflow.com/a/49587869/210550